integ-test/src/test/java/org/opensearch/sql/sql/NestedIT.java

```diff
@@ -366,4 +368,34 @@ public void test_nested_in_where_as_predicate_expression_with_relevance_query()
     assertEquals(1, result.getInt("total"));
     verifyDataRows(result, rows(10, "a"));
   }
+
+  @Test
+  public void nested_function_with_date_types_as_object_arrays_within_arrays_test() {
+    String query = "SELECT nested(address.moveInDate) FROM " + TEST_INDEX_NESTED_SIMPLE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(11, result.getInt("total"));
+    verifyDataRows(result,
```

Comment:
what would be the data type of nested(address.moveInDate), object or array?

Reply:
It would be of type object. address would be of type nested, I'll add the schema verification.

Reply:
i see. one more question. 
some rows is JSONObject, some rows are JSONArray, how does JDBC handle it?

Reply:
In some cases this breaks JDBC, but there is no change in the expected output with this PR. An example of this would be in `ObjectFieldSelectIT` where we do a select on an `OBJECT` type array, and a `NESTED` type array.

[ObjectFieldSelectIT.java#L69-L93](https://github.com/opensearch-project/sql/blob/main/integ-test/src/test/java/org/opensearch/sql/legacy/ObjectFieldSelectIT.java#L69-L93)
[people.json](https://github.com/opensearch-project/sql/blob/main/doctest/test_data/people.json)
[people-mapping](https://github.com/opensearch-project/sql/blob/main/doctest/test_mapping/people.json#L18-L30)

When selecting on an array that is not of type `NESTED`, the first value in the array is returned and is functional with JDBC. As shown by the query:
```SQL
SELECT accounts FROM people;
```

While on the other hand the query that does a basic SELECT on an `NESTED` type array returns the whole array which breaks JDBC.

Query:
```SQL
SELECT projects FROM people;
```

A user can get around this by using the `nested` function to flatten the response for a correct JDBC output.